src/main/java/ch/njol/skript/expressions/ExprLocationVectorOffset.java

```diff
@@ -70,17 +77,44 @@ public boolean isSingle() {
 	public Class<? extends Location> getReturnType() {
 		return Location.class;
 	}
-
-	@Override
+  
 	public Expression<? extends Location> simplify() {
 		if (location instanceof Literal<Location> && vectors instanceof Literal<Vector>)
 			return SimplifiedLiteral.fromExpression(this);
 		return this;
 	}
 
 	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		return location.toString() + " offset by " + vectors.toString();
+	public String toString(@Nullable Event event, boolean debug) {
+		return new SyntaxStringBuilder(event, debug)
+			.append(location)
+			.append("offset by")
+			.append(vectors)
+			.append(usingLocalAxes ? "using local axes" : "")
+			.toString();
+	}
+
+	/**
+	 * Returns a location offset from the given location, adjusted for the location's rotation.
+	 * <p>
+	 * This behaves similarly to Minecraft's {@code /summon zombie ^ ^ ^1} command,
+	 * where the offset is applied relative to the entity's facing direction.
+	 *
+	 * @see <a href="https://minecraft.wiki/w/Coordinates#Local_coordinates">Local Coordinates</a>.
+	 * @param loc The location
+	 * @param offset The offset
+	 * @return The offset location
+	 */
+	private static Location getFacingRelativeOffset(Location loc, Vector offset) {
+		float yawRad = (float) Math.toRadians(-loc.getYaw());
+		float pitchRad = (float) Math.toRadians(loc.getPitch());
+		float rollRad = 0f;
+
+		Quaternionf rotation = new Quaternionf().rotateYXZ(yawRad, pitchRad, rollRad);
+		Vector3f localOffset = offset.toVector3f();
+		rotation.transform(localOffset);
+
+		return loc.add(localOffset.x, localOffset.y, localOffset.z);
```

Comment:
it'd make the code a bit nicer and standard if this returned a vector instead of a location, so the .add is done in the same spot as the normal add.